lib/modules/ipfs/index.js

```diff
@@ -66,20 +58,31 @@ class IPFS {
         }
         self.logger.trace("IPFS available");
         return cb({name: "IPFS ", status: 'on'});
-      }
+      });
     });
   }
 
-  addProviderToEmbarkJS() {
-    const self = this;
-    // TODO: make this a shouldAdd condition
-    if (this.storageConfig === {}) {
-      return;
+  _getNodeUrl() {
+    if (this.storageConfig.upload.provider === 'ipfs') {
+      let {protocol, host, port} = this.storageConfig.upload;
+      return (protocol || 'http') + '://' + host + ':' + port + '/api/v0/version';
     }
 
-    if (this.storageConfig.available_providers.indexOf('ipfs') < 0 || _.findWhere(this.storageConfig.dappConnection, {'provider': 'ipfs'}) === undefined || this.storageConfig.enabled !== true) {
-      return;
+    for (let connection of this.storageConfig.dappConnection) {
+      if (connection.provider === 'ipfs') {
```

Comment:
`connection` might === `$BZZ`, and so `connection.provider` might throw?

Reply:
What about the case when the user has specified an upload provider as well as a dappConnection provider that are both ipfs, but point to different ipfs urls? Maybe this use case is not supported?

Reply:
It is not, there can be only node ipfs node, so this will use the first one available